src/modules/RA/DataLayer/components/tabs/StyleTab/Style/PieChartRadiusStyleField.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import { makeStyles } from '@material-ui/core/styles';
+import { useField } from 'react-final-form';
+import { SelectInput, NumberInput, required } from 'react-admin';
+import { fieldTypes } from '../../../../../DataSource';
+
+import Condition from '../../../../../../../components/react-admin/Condition';
+import FieldOption from '../../FieldOption';
+
+import styles from './styles';
+
+const isRequired = [required()];
+
+const useStyles = makeStyles(styles);
+
+const PieChartRadiusStyleField = ({ path, fields }) => {
+  const classes = useStyles();
+
+  const {
+    input: { value: type },
+  } = useField(`${path}.type`);
+
+  if (type === 'none') {
+    return null;
+  }
+
+  return (
+    <div className={classes.styleField}>
+      <Condition when={`${path}.type`} is="fixed">
+        <NumberInput
+          source={`${path}.value`}
+          label="style-editor.fixed.diameter"
+        />
+      </Condition>
+
+      <Condition when={`${path}.type`} is="variable">
+        {fields && (
+          <>
+            <SelectInput
+              source={`${path}.field`}
+              helperText="style-editor.field-help"
+              style={{ minWidth: '20em', margin: '1em 0' }}
+              label="style-editor.field"
+              validate={isRequired}
+              optionText={<FieldOption />}
+              choices={fields
```

Comment:
It might be worth to put this in a `useMemo` to avoid the "re-creation" of two `Array` from the `map().filter()` at each render.